File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        area=0
        l=0
        r=len(height)-1
        while l<r:
            length=min(height[l],height[r])
            area=max(area,length*(r-l))
            if length==height[l]:
                l+=1
            else:
                r-=1
        return area
            

File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int l = 0, r = height.length-1;
        int maxArea = 0;
        while(l<r){
            if(height[l]<height[r]){
                maxArea = Math.max(maxArea,height[l]*(r-l));
                l++;
            }
            else if(height[l]>height[r]){
                maxArea = Math.max(maxArea, height[r]*(r-l));
                r--;
            }
            else{
                maxArea = Math.max(maxArea, height[r]*(r-l));
                l++;
                r--;
            }
        }
        return maxArea;
    }
}